java/ql/lib/semmle/code/java/frameworks/JaxWS.qll

```diff
@@ -955,3 +961,17 @@ private class VulnerableEntity extends XssSinkBarrier {
       ).getArgument(0)
   }
 }
+
+/**
+ * Model sources stemming from `ContainerRequestContext`.
+ */
+private class ContainerRequestContextModel extends SourceModelCsv {
+  override predicate row(string s) {
+    s =
+      ["javax", "jakarta"] + ".ws.rs.container;ContainerRequestContext;true;" +
+        [
+          "getAcceptableLanguages", "getAcceptableMediaTypes", "getCookies", "getEntityStream",
```

Comment:
Can `Locale` contain arbitrary taint?

Reply:
As it turns out yes -- the docs for Locale specify what the parameters should be (e.g. `language - An ISO 639 alpha-2 or alpha-3 language code`), but it doesn't actually check that that's the case. I forget where exactly but I think @atorralba wrote a little test case recently showing attack code from an HTTP Accept header making it through unaltered to Locale.toString.

Reply:
Yep https://github.com/github/codeql/pull/6478#discussion_r690274205